### PR TITLE
Add UA based rewrites example

### PR DESCRIPTION
**Pull request by goncy**
This example shows how to load a different page based on users UA

image

Deploy: https://ua-based-app-goncy-vercelcom.vercel.app/

### CODE REVIEW

File: edge-functions/ua-based-app/components/Home.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+          different scripts, styles, etc. We want to decide which page to load
+          based on users UA without loading unnecesary assets for the current
+          viewport.
+        </Text>
+      </section>
+
+      <section className="flex flex-col gap-6 mt-12">
+        <Text variant="h2">Folder structure</Text>
+        <Text>
+          We will rewrite our user to different pages based on its UA so we have
```
Comment (gbibeaul) [inline edge-functions/ua-based-app/components/Home.tsx:34]: Maybe adding a link to a UA description could help educate people who are unfamiliar with the concept.

**Review by gbibeaul — APPROVED** [review]
Tiny request and language corrections

File: edge-functions/ua-based-app/pages/_middleware.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+export function middleware(req: NextRequest) {
+  // Clone the URL
+  const url = req.nextUrl.clone()
+
+  // Prevent internals from being accessed canonically
+  if (url.pathname.startsWith(`/_viewport`)) {
+    return new Response(null, { status: 404 })
+  }
+
```
Comment (lfades) [inline edge-functions/ua-based-app/pages/_middleware.ts:13]: We should also check for public files

File: edge-functions/ua-based-app/public/UA example - Implementation (1).jpg:Zone.Identifier [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[ZoneTransfer]
+ZoneId=3
+HostUrl=about:internet
```
Comment (lfades) [inline edge-functions/ua-based-app/public/UA example - Implementation (1).jpg:Zone.Identifier:3]: What are these?

File: edge-functions/ua-based-app/public/UA example - Implementation (2) (1).jpg:Zone.Identifier [hunk 4]
```diff
@@ -0,0 +1,3 @@
+[ZoneTransfer]
+ZoneId=3
+HostUrl=about:internet
```
Comment (lfades) [inline edge-functions/ua-based-app/public/UA example - Implementation (2) (1).jpg:Zone.Identifier:3]: What are these?

File: edge-functions/ua-based-app/README.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
+# ua-based-app example
```
Comment (lfades) [inline edge-functions/ua-based-app/README.md:1]:
```suggestion 
# User Agent Based Rendering
```

**Review by lfades — CHANGES_REQUESTED** [review]
The example should have a more clear name, like: `user-agent-based-rendering`, or `render-different-viewport-for-user-agent`. Whatever makes more sense but that is clear enough

File: edge-functions/ua-based-app/public/UA example - Implementation (1).jpg:Zone.Identifier (hunk 3 above)
Comment (goncy) [reply edge-functions/ua-based-app/public/UA example - Implementation (1).jpg:Zone.Identifier:3]: Ah, probably squoosh generated those

**Review by lfades — APPROVED** [review]
Added some changes:

- The 404 now renders the 404 page
- The public files check now is an empty return, returning the request is no good
- Changed the contents of the main page to be about explaining what's the example about instead of giving a tutorial

**Pull request merged**